pkg/git/providers/git/git.go

```diff
@@ -0,0 +1,71 @@
+package git
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+const (
+	GitPasswordTokenEnv       = "GIT_PASSWORD"
+	EksaGitPasswordTokenEnv   = "EKSA_GIT_PASSWORD"
+	GitPrivateKeyTokenEnv     = "GIT_PRIVATE_KEY"
+	EksaGitPrivateKeyTokenEnv = "EKSA_GIT_PRIVATE_KEY"
+)
+
+type Options struct {
+	RepositoryUrl string
+	User          string
+}
+
+func GetGitPasswordFromEnv() (string, error) {
+	password, err := getTokenFromEnv(EksaGitPasswordTokenEnv, GitPasswordTokenEnv)
+	if err != nil {
+		return "", err
+	}
+
+	return password, nil
+}
+
+func GetGitPrivateKeyFileFromEnv() (string, error) {
+	privateKeyFilePath, err := getTokenFromEnv(EksaGitPrivateKeyTokenEnv, GitPrivateKeyTokenEnv)
+	if err != nil {
+		return "", err
+	}
+	privateKeyfileExistsAndIsNotEmpty := validateFilePath(privateKeyFilePath)
+	if !privateKeyfileExistsAndIsNotEmpty {
+		logger.V(4).Info("Private key file does not exist or is empty", "file path", privateKeyFilePath)
+		return "", fmt.Errorf("private key file does not exist at %s or is empty", privateKeyFilePath)
+	}
+
+	return privateKeyFilePath, nil
+}
+
+func getTokenFromEnv(EksaEnvVar, GitEnvVar string) (string, error) {
+	logger.V(4).Info("Checking for environment variable", "env var", EksaEnvVar)
+
+	env := make(map[string]string)
+	val, ok := os.LookupEnv(EksaEnvVar)
+	if !ok || len(val) == 0 {
+		logger.Info("%s environment variable is not set", EksaEnvVar)
+		return "", fmt.Errorf("environment variable %s is not set", EksaEnvVar)
+	}
+	if len(val) > 0 {
```

Comment:
I don't think we need to do any of this for the `EKSA_GIT_PRIVATE_KEY` env var, right? we just need to get the path value as a string.

Reply:
Added a validation on the path to see if it exists and the file is not empty.